src/libcore/alloc.rs

```diff
@@ -140,6 +140,27 @@ impl Layout {
         unsafe { Layout::from_size_align_unchecked(size, align) }
     }
 
+    /// Produces layout describing a record that could be used to
+    /// allocate backing structure for `T` (which could be a trait
+    /// or other unsized type like a slice).
+    ///
+    /// # Safety
+    ///
+    /// This function is only safe to call with a pointer to a valid allocation.
+    /// In other words, if it would be safe to dereference the pointer as a
+    /// [`*const MaybeUninit<T>`][mem::MaybeUninit], then this function is safe.
```

Comment:
Wait, you want to require the pointer to be dereferencable? That's much stronger than anything we discussed, I feel, and also what is even the point of this method?

Reply:
Note that this is only the current, most conservative starting place. [I did mention the intent here](https://github.com/rust-lang/rust/pull/69079#issuecomment-593205078). Note that the use of `MaybeUninit` here is intending to make it such that the pointee does not need to be initialized, just exist.

Reply:
This is quite confusing though as `MaybeUninit<T>` does not exist unless `T` is sized.

Reply:
Alright, I've got a looser definition closer to @SimonSapin's that should be clearer now.